File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        
        cost.append(0)
        n = len(cost)
        dp = [0]*(n+2)
        def helper(x):
            if x <=0:
                return 0
            if dp[x]:
                return dp[x]
            
            dp[x] = min( helper(x-1), helper(x-2))+ cost[x-1]
            return dp[x]
            
            
        helper(n)
        return dp[-2]
            
            

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
            
        
        
            int n = cost.size() ; 
            int one = cost[0] ; 
            int two = cost[1] ; 
            if(n<=2) return min(one,two) ;
        
            int three ; 
            for(auto i = 2; i <n ; i++){
                three = cost[i] + min(one,two) ; 
                one = two ; 
                two = three ; 
            }
            return min(three,one) ; 


    }
};